src/core/ComboBox/ComboBox.tsx

```diff
@@ -301,7 +301,9 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
           },
           // ComboBox.MenuItem handles data-iui-index and iui-focused through context
           // but we still need to pass them here for backwards compatibility with MenuItem
-          className: cx(customItem.props.className, 'iui-focused'),
+          className: cx(customItem.props.className, {
+            'iui-focused': focusedIndex === __originalIndex,
```

Comment:
now this got weird :D you said yourself, that you have early return, if focusedIndex would equal to original one

Reply:
i've read this comment about 10 different times in 4 days and i'm still not sure what it means 😄

Reply:
Oh, I think previously there wasnt (or I didnt notice) check for selectedIndex too.
So when I commented, idea was:
You already check if focused index is not equal to original and early return. So on that particalar code line `focused === __originalIndex` could be replaced just with `true` as it was before.